server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -145,12 +150,31 @@ private RepositoriesMetadata buildRepositoriesMetadata(DiscoveryNode node) {
         return new RepositoriesMetadata(repositoryMetadataList);
     }
 
+    private void validateDataRepositoryAttributes(DiscoveryNode node) {
+        if (node.getAttributes().containsKey(REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY)
+            || node.getAttributes().containsKey(REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY)) {
+            validateAttributeNonNull(node, REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY);
+            validateAttributeNonNull(node, REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY);
```

Comment:
Shouldn't we check for cluster state repository as well? As I understand, cluster state repository can be set even if data repositories are not set but when data repositories are set, cluster state repository needs to be present, right?

Reply:
Data persistence does not have a dependency on cluster state persistence. Even when user trigger restore API, the metadata is used from the in-memory cluster state.
If we want to enforce the user to also have remote state enabled when remote data store is enabled, then I can add this validation.

Reply:
Added validation to have cluster state repo as well.